src/sentry/notifications/utils/tasks.py

```diff
@@ -85,6 +102,13 @@ def _send_notification(notification_class_name: str, arg_list: Iterable[Mapping[
                 output_kwargs[arg["key"]] = instance
             else:
                 output_args.append(instance)
+        elif arg["type"] == "lazyobject":
+            arg = RpcUser.parse_obj(arg["data"])
+
+            if arg["key"]:
+                output_kwargs[arg["key"]] = SimpleLazyObject(lambda: instance)
+            else:
+                output_args.append(SimpleLazyObject(lambda: instance))
```

Comment:
We could likely skip building the `SimpleLazyObject` and only re-create the `RpcUser`.